### PR TITLE
Tidy up Promise tab

### PR DESCRIPTION
**Pull request by nummi**
`@, this, </>, action`,etc.

### CODE REVIEW

File: app/templates/promise-tree-toolbar.hbs [hunk 1]
```diff
@@ -23,26 +23,26 @@
   <Ui::ToolbarDivider />
 
   <button
-    class="{{if noFilter "active"}} toolbar__radio toolbar__radio--first js-filter"
-    {{action "setFilter" "all"}}
+    class="{{if (eq this.filter "all") "active"}} toolbar__radio toolbar__radio--first js-filter"
```
Comment (rwwagner90) [inline app/templates/promise-tree-toolbar.hbs]: 🔥

File: app/templates/components/promise-item.hbs [hunk 2]
```diff
@@ -1,76 +1,76 @@
-{{#list.cell
-  class=(concat "list__cell_main " expandedClass)
-  style=labelStyle
-  on-click=(action toggleExpand model)
+{{#@list.cell
```
Comment (rwwagner90) [inline app/templates/components/promise-item.hbs]: Was this `@` on purpose? I suppose it is valid, but looks odd having `#@`

File: app/templates/promise-tree.hbs [hunk 3]
```diff
@@ -9,18 +9,18 @@
...
-          tracePromise=(action "tracePromise")
-          inspectObject=(action "inspectObject")
-          sendValueToConsole=(action "sendValueToConsole")
-          list=list
-        }}
+        <PromiseItem
+          @model={{content}}
+          @filter={{this.filter}}
+          @effectiveSearch={{this.effectiveSearch}}
+          @toggleExpand={{fn this.toggleExpand content}}
```
Comment (rwwagner90) [inline app/templates/promise-tree.hbs]: I'm curious why only two of the actions needed `content` passed?

File: app/templates/components/promise-item.hbs (hunk 2 above)
Comment (nummi) [reply app/templates/components/promise-item.hbs]: It looks terrible :-P. But `list` is passed in that way, right? Maybe we should get a consult from an expert?

File: app/templates/promise-tree.hbs (hunk 3 above)
Comment (nummi) [reply app/templates/promise-tree.hbs]: I like to keep the components as dumb as possible but maybe consistency should win here.

File: app/templates/components/promise-item.hbs (hunk 2 above)
Comment (rwwagner90) [reply app/templates/components/promise-item.hbs]: Does angle bracket syntax work here?

File: app/templates/promise-tree.hbs (hunk 3 above)
Comment (rwwagner90) [reply app/templates/promise-tree.hbs]: If only two need it, that's fine. I just wanted some clarity.

File: app/templates/components/promise-item.hbs (hunk 2 above)
Comment (nummi) [reply app/templates/components/promise-item.hbs]: Yep :+1:

Comment (rwwagner90) [reply app/templates/components/promise-item.hbs]: Nice! That looks much better 😃

File: app/templates/promise-tree.hbs [hunk 4]
```diff
@@ -9,18 +9,18 @@
...
-          model=content
-          filter=filter
-          effectiveSearch=effectiveSearch
-          toggleExpand=(action "toggleExpand")
-          tracePromise=(action "tracePromise")
-          inspectObject=(action "inspectObject")
-          sendValueToConsole=(action "sendValueToConsole")
-          list=list
-        }}
+        <PromiseItem
```
Comment (stefanpenner) [inline app/templates/promise-tree.hbs]: Fancy!

Comment (nummi) [reply app/templates/promise-tree.hbs]: Thanks. 😆

**Comment by nummi** [comment]
code climate 🤷‍♂

**Pull request merged**